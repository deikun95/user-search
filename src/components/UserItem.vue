<template>
  <div class="wrapper">
    <li class="item" @click="dialogHandler">
      <img :src="user.avatar_url" alt class="img" />
      <div class="name">
        <span class="username">{{user.login}}</span>
      </div>
    </li>
    <UserDialog :open="open" @close="handleDialogClose" :user="user" />
  </div>
</template>

<script>
import UserDialog from "./UserDialog";
export default {
  name: "UserItem",
  components: {
    UserDialog
  },
  data() {
    return {
      open: false
    };
  },
  props: {
    user: {
      type: Object,
      default: () => ({})
    }
  },
  methods: {
    dialogHandler: function() {
      this.open = true;
    },
    handleDialogClose() {
      this.open = false;
    }
  }
};
</script>

<style lang="scss">
.item {
  display: flex;
  // justify-content: space-between;
  align-items: center;
  width: 400px;
  height: 80px;
  border: 2px solid rgb(235, 235, 235);
  margin-bottom: 30px;
  border-radius: 50px;
  box-shadow: 0 1px 10px 0 rgba(0, 0, 0, 0.1);
  cursor: pointer;
  user-select: none;
  &:hover {
    background-color: rgba(235, 235, 235, 0.4);
  }
}
.img {
  display: flex;
  // justify-content: flex-start;
  border-radius: 50%;
  width: 80px;
  height: 80px;
  background-image: url();
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}
.name {
  margin: auto;
}
.username {
  font-size: 20px;
}
</style>