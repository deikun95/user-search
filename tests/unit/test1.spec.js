import { shallowMount } from '@vue/test-utils'
import UserItem from '@/components/Useritem'
describe('UserItem', () => {
  it('renders the component', () => {
    const wrapper = shallowMount(UserItem)
    expect(wrapper.html()).toMatchSnapshot()
  })
})