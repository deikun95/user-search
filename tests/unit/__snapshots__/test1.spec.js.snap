// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem renders the component 1`] = `
<div class="wrapper">
  <li class="item"><img alt="" class="img">
    <div class="name"><span class="username"></span></div>
  </li>
  <userdialog-stub user="[object Object]"></userdialog-stub>
</div>
`;
